lib/archethic/p2p/supervisor.ex

```diff
@@ -11,11 +11,20 @@ defmodule Archethic.P2P.Supervisor do
 
   use Supervisor
 
+  @table_name :connection_status
+
   def start_link(args \\ []) do
     Supervisor.start_link(__MODULE__, args, name: Archethic.P2PSupervisor)
   end
 
   def init(_args) do
+    # Start an ets table to manage node connection status.
+    # This reduce the number of message sent to the Connection GenServer
+    if :ets.whereis(@table_name) == :undefined do
```

Comment:
If we want to be cleaner, the best would be to isolate the table in a genserver which might serve more capability and will support hot reload, if we need to change the state in the ets table easily. 
This gen server could be dedicated to connection information or status and might manage its own table and lifecycle.